### PR TITLE
Fixed NaN comparison

### LINKED ISSUE #644: `NaN < 0` errors
I would expect that `NaN < 0` be `false`, but instead it gives

```
>>> NaN < 0.0
error: runtime error
 = Conversion error: unit '' can not be converted to ''
```

### PR DESCRIPTION
**Pull request by rben01**
All comparisons involving at least one NaN now return false, regardless of units

Closes #644

### CODE REVIEW

**Comment by rben01** [comment]
One question is whether NaN should be comparable with other quantities regardless of units. I think yes, it is like 0 in this regard: the answer is known to be false regardless of units.

File: numbat/src/quantity.rs [hunk 1]
```diff
@@ -347,7 +345,36 @@ impl PrettyPrint for Quantity {
     }
 }
 
+pub(crate) enum QuantityOrdering {
+    IncompatibleUnits,
+    NanOperand,
+    Less,
+    Equal,
+    Greater,
+}
```
Comment (sharkdp) [inline numbat/src/quantity.rs:354]: Would this lead to slightly less code?
```suggestion
pub(crate) enum QuantityOrderingResult {
    IncompatibleUnits,
    NanOperand,
    Ok(std::cmp::Ordering),
}
```

File: numbat/src/quantity.rs [hunk 2]
```diff
@@ -347,7 +345,36 @@ impl PrettyPrint for Quantity {
...
+            return QuantityOrdering::IncompatibleUnits;
+        };
+
+        match self.value.partial_cmp(&other_converted.value) {
+            Some(cmp) => match cmp {
+                std::cmp::Ordering::Less => QuantityOrdering::Less,
+                std::cmp::Ordering::Equal => QuantityOrdering::Equal,
+                std::cmp::Ordering::Greater => QuantityOrdering::Greater,
+            },
+            None => unreachable!("unexpectedly got a None partial_cmp from non-NaN arguments"),
```
Comment (sharkdp) [inline numbat/src/quantity.rs:374]: Maybe catch this with a `.expect(…)` above, using the same message?

**Comment by sharkdp** [comment]
> One question is whether NaN should be comparable with other quantities regardless of units. I think yes, it is like 0 in this regard: the answer is known to be false regardless of units.

Yes, we want `NaN` to be polymorphic. If `0` is polymorphic, then so is `0 × inf == NaN`:
```
meter × NaN = meter × (0 × inf) = (meter × 0) × inf = 0 × inf = NaN
```
(ignoring associativity-violations in FP math for a second)

If `NaN` wouldn't be polymorphic, that equation above would not by typeable (`Length × Scalar = Scalar`?).

**Comment by rben01** [comment]
Not sure how I accidentally closed this or why I ended up force pushing, or why the macOS job didn't complete, but this PR should be good now.

**Review by sharkdp — APPROVED** [review]
Thank you!

**Pull request merged**